trio/_core/_entry_queue.py

```diff
@@ -43,7 +50,9 @@ async def task(self):
         #     https://bugs.python.org/issue13697#msg237140
         assert self.lock.__class__.__module__ == "_thread"
 
-        def run_cb(job):
+        def run_cb(
+            job: tuple[Callable[..., object], Iterable[object]]
+        ) -> Literal[True]:
```

Comment:
Nothing ever uses this return value, might as well just remove it while we're here.